shared/fs/index.js

```diff
@@ -1,30 +1,119 @@
-import {Box, Button, Text, ButtonBar} from '../common-adapters'
-
-type Props = {
-  counter: number,
-  you: ?string,
-  increase: () => void,
-  increase10: () => void,
-}
-
-const Fs = ({counter, you, increase, increase10}: Props) => (
-  <Box style={containerStyle}>
-    <Text type="Header">Hi {you}!</Text>
-    <Text type="BodySemibold">Count: {counter}</Text>
-    <ButtonBar align="flex-start">
-      <Button type="Primary" onClick={increase} label="Up by 1" />
-      <Button type="Primary" onClick={increase10} label="Up by 10" />
-    </ButtonBar>
-  </Box>
-)
+import * as Types from '../constants/types/fs'
+import {globalStyles, globalColors, globalMargins, isMobile} from '../styles'
+import {Box, BackButton, Button, ClickableBox, Icon, List, Text} from '../common-adapters'
+import {type IconType} from '../common-adapters/icon'
+
+const stylesCommonCore = {
+  alignItems: 'center',
+  borderBottomColor: globalColors.black_05,
+  borderBottomWidth: 1,
+  justifyContent: 'center',
+}
 
-const containerStyle = {
+const stylesCommonRow = {
+  ...globalStyles.flexBoxRow,
+  ...stylesCommonCore,
+  minHeight: isMobile ? 64 : 48,
+  padding: 8,
+}
+
+const stylesContainer = {
   ...globalStyles.flexBoxColumn,
+  ...globalStyles.fullHeight,
   flex: 1,
-  padding: globalMargins.xlarge,
 }
 
-export default Fs
+type FolderHeaderProps = {
+  title: string,
+}
+
+type FileRowProps = {
+  path: Types.Path,
+  icon: IconType,
+  showFileData: () => void,
+}
+
+type FolderProps = {
+  name: string,
+  path: Types.Path,
+  visibility: Types.Visibility,
+  items: Array<string>,
+  onViewFolder: (p: Path) => void,
+  onViewFile: (p: Path) => void,
+}
+
+const FolderBoxStyle = {...globalStyles.flexBoxColumn, flex: 1, justifyContent: 'stretch'}
+
+const styleOuterContainer = {
+  position: 'relative',
+  height: '100%',
+}
+
+const FolderHeader = ({title}: FolderHeaderProps) => (
+  <Box>
+    <Box style={{...stylesCommonRow, alignItems: 'center', borderBottomWidth: 0}}>
+      <Text type="HeaderBig" style={{padding: globalMargins.xtiny}}>
+        {title}
+      </Text>
+    </Box>
+  </Box>
+)
+
+const FileRow = ({name, path, icon, showFileData}: FileRowProps) => {
+  return (
+    <ClickableBox onClick={showFileData} style={{...stylesCommonRow}}>
+      <Box key={name} style={{...globalStyles.flexBoxRow, alignItems: 'center', flex: 1}}>
+        <Icon type={icon} style={{marginRight: globalMargins.small}} />
+        <Box style={FolderBoxStyle}>
+          <Text type="Body">
+            {name}
+          </Text>
+        </Box>
+      </Box>
+    </ClickableBox>
+  )
+}
+
+const iconTypes : IconType = {
+  folder: isMobile ? 'icon-folder-private-24' : 'icon-folder-private-24',
+  file: isMobile ? 'icon-file-24' : 'icon-file-24',
+}
+
+export class Files extends React.PureComponent<FolderProps> {
+  _renderRow = (index, item) => {
+    const iconType = iconTypes[item.type]
+    const showFileData = () => {
```

Comment:
we should avoid creating closures like this in what is essentially a render function. Doing this will cause the showfileData prop to change on every render causing further rerenders down the line. Likely better is to keep it accepting a param and have FileRow call it with the path itself
aka
```
showFileData={item.type === 'folder' ? this.props.onViewFolder : this.props.onViewFile}
```
```